onnxruntime/test/providers/qnn/conv_test.cc

```diff
@@ -397,218 +410,328 @@ TEST_F(QnnCPUBackendTests, ConvTranspose1Df32_DynamicWeights_DefaultBias) {
+                                     "NOTSET",
+                                     ExpectedEPNodeAssignment::All);
+}
+
+// Tests 16-bit QDQ Conv with dynamic weights and bias (uses QNN's Conv2d)
+// TODO(adrianlizarraga): Inaccuracy detected for output 'output', element 0.
+// Output quant params: scale=6585.83056640625, zero_point=65535.
+// Expected val: -67.35833740234375
+// QNN QDQ val: -431602400 (err 431602336)
+// CPU QDQ val: 0 (err 67.35833740234375)
+TEST_F(QnnHTPBackendTests, DISABLED_ConvU16S16S32_DynamicBias) {
+  RunHTPConvOpTest<uint16_t, int16_t>("Conv",
+                                      TestInputDef<float>({1, 1, 5, 5}, false, -10.0f, 10.0f),  // Input
+                                      TestInputDef<float>({2, 1, 3, 3}, false, -10.0f, 10.0f),  // Weights
+                                      TestInputDef<float>({1}, false, {2.0f}),                  // Bias
+                                      {1, 1},                                                   // Strides
+                                      {0, 0, 0, 0},                                             // Pads
+                                      {1, 1},                                                   // Dilations
+                                      "NOTSET",
+                                      ExpectedEPNodeAssignment::All,
+                                      true);  // Use com.microsoft QDQ ops for 16-bit
+}
+
+// Tests 16-bit QDQ Conv with dynamic weights and no bias (uses QNN's DepthWiseConv2d)
+// TODO(adrianlizarraga): FAIL: Failed to finalize QNN graph. Error code 1002
+TEST_F(QnnHTPBackendTests, DISABLED_ConvU16S16S32_NoBias) {
+  RunHTPConvOpTest<uint16_t, int16_t>("Conv",
+                                      TestInputDef<float>({1, 1, 5, 5}, false, -10.0f, 10.0f),  // Input
+                                      TestInputDef<float>({1, 1, 3, 3}, false, -10.0f, 10.0f),  // Weights
+                                      TestInputDef<float>(),                                    // Bias
+                                      {1, 1},                                                   // Strides
+                                      {0, 0, 0, 0},                                             // Pads
+                                      {1, 1},                                                   // Dilations
+                                      "NOTSET",
+                                      ExpectedEPNodeAssignment::All,
+                                      true);  // Use com.microsoft QDQ ops for 16-bit
+}
+
+// Tests 16-bit activations, 8-bit static weights QDQ Conv with static bias.
+// TODO(adrianlizarraga): Inaccuracy detected for output 'output', element 2.
+// Output quant params: scale=0.0041189868934452534, zero_point=16353.
+// Expected val: 202.57948303222656
+// QNN QDQ val: 201.96627807617188 (err 0.6132049560546875)
+// CPU QDQ val: 202.028076171875 (err 0.5514068603515625)
```

Comment:
Are these diffs still a concern? I 'm guessing not given the test is enabled.

Reply:
These diffs are a concern, and we will be working with Qualcomm to investigate. However, since the difference is not "too large" I would like to enable it with a larger-than-normal tolerance so that we have some enabled tests that run 16bit QDQ convs on QNN EP.

I have plans to overhaul the way we specify allowable tolerances for QNN EP tests, but I would like to handle that separately.

Reply:
Those seem reasonable for floating point calculations in that range.

Reply:
Yeah, in the past we investigated cases in which the Qnn output was 1 scale's worth away from the ORT output value (i.e., 1 quantization unit). This was reasonably rare for 8bit qdq. Now, we're going to need to rethink this moving forward. Will handle in a future PR.